File: receptor.py
from cmath import log
import os
import subprocess


def validar(arg):
    dir = os.path.expanduser('~')
    path = arg.replace(dir, "http://localhost:8000")
    transTable = path.maketrans("\\", "/")
    path = path.translate(transTable)
    return path


def getdefaultBrow():
    try:
        from winreg import HKEY_CLASSES_ROOT, HKEY_CURRENT_USER, OpenKey, QueryValueEx

        with OpenKey(HKEY_CURRENT_USER, r'SOFTWARE\Microsoft\Windows\Shell\Associations\UrlAssociations\http\UserChoice') as regkey:
            browser_choice = QueryValueEx(regkey, 'ProgId')[0]

        with OpenKey(HKEY_CLASSES_ROOT, r'{}\shell\open\command'.format(browser_choice)) as regkey:
            browser_path_tuple = QueryValueEx(regkey, None)
            browser_path = browser_path_tuple[0].split('"')[1]

        return browser_path

    except Exception:
        log.error(
            'Failed to look up default browser in system registry. Using fallback value.')


def open(link, browser, df, solo):
    if solo == 1:
        link = "--app="+link

    if df:
        p = subprocess.Popen([getdefaultBrow(), link])
    else:
        p = subprocess.Popen([browser, link])


def ini(ruta, browser, df, solo):
    if len(ruta) > 1:
        path = "https://www.photopea.com/#{\"files\":[\"" + \
            validar(ruta[1]) + "\"]}"
        open(path, browser, df, solo)
        print(":)?")
    else:
        open("https://www.photopea.com", browser, df, solo)

File: SourceCode/PhotOp_Java/src/photop/Receptor.java
package photop;

import java.io.File;
import java.io.IOException;
import java.text.ParseException;
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 *
 * @author Señor C.
 */
public class Receptor {

    private static String validate(String dir) {
        return dir.replace(System.getProperty("user.home"),
                "http://localhost:8000").replace("\\", "/");
    }

    private static Scanner getScanner(String query) throws IOException {
        Process process = Runtime.getRuntime().exec(query);
        return new Scanner(process.getInputStream());
    }

    private static String getProdId(String query) throws IOException {
        Scanner sc = getScanner(query);
        query = sc.nextLine().trim();
        query = sc.nextLine().trim();
        query = sc.nextLine().trim();
        sc.close();
        String[] vec = query.replace(" ", ",").split(",");
        return vec[vec.length - 1];
    }

    private static String findPath(Scanner sc, String aux) {
        while (sc.hasNextLine()) {
            aux = sc.nextLine();
            Matcher matcher = Pattern.compile("[\"](.+?)[\"]").matcher(aux);
            if (matcher.find()) {
                sc.close();
                aux = matcher.group(1);
                return aux;
            }
        }
        sc.close();
        return "Error: browser path missed";
    }

    public static String getDefaultBrowser() {
        try {
            String prodId = getProdId("REG QUERY HKEY_CURRENT_USER\\SOFTWARE\\"
                    + "Microsoft\\Windows\\Shell\\Associations\\UrlAssociations"
                    + "\\http\\UserChoice");
            Scanner kb = getScanner("REG QUERY HKEY_CLASSES_ROOT\\" + prodId
                    + "\\shell\\open\\command");
            return findPath(kb, prodId);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return "Error: Unable to get default browser";
    }

    public static void open(String link, String browser, boolean df, boolean solo) {
        try {
            if (solo) {
                link = "--app=" + link;
            }
            if (df) {
                Process p = Runtime.getRuntime().exec(getDefaultBrowser() + " " + link);
            } else {
                String aux = getrealPath(browser);
                Process process = new ProcessBuilder(aux, link).start();
            }
        } catch (Exception e) {

        }
    }

    public static String getrealPath(String path) throws IOException, ParseException {
        if (path.endsWith(".lnk")) {
            final WindowsShortcut sc = new WindowsShortcut(new File(path));
            return sc.getRealFilename();
        }
        return path;
    }

    public static void ini(String[] ruta, String browser, boolean df, boolean solo) {
        if (ruta.length > 0) {
            String files = "";
            for (int i = 0; i < ruta.length; i++) {
                System.out.println(ruta[i]);
                if (i > 0) {
                    files += ",";
                }
                files += "\\\"" + validate(ruta[i]) + "\\\"";
            }
            final String path = "https://www.photopea.com/#{\\\"files\\\":[" + files + "]}";
            open(path, browser, df, solo);
        } else {
            open("https://www.photopea.com", browser, df, solo);
        }

    }

}
